packages/next-swc/crates/next-build-test/src/main.rs

```diff
@@ -106,18 +107,16 @@ fn main() {
                             subscriber.with(EnvFilter::builder().parse(trace).unwrap());
                         let trace_file = "trace.log";
                         let trace_writer = std::fs::File::create(trace_file).unwrap();
-                        let (trace_writer, guard) = TraceWriter::new(trace_writer);
+                        let (trace_writer, trace_writer_guard) = TraceWriter::new(trace_writer);
                         let subscriber = subscriber.with(RawTraceLayer::new(trace_writer));
 
-                        let guard = ExitGuard::new(guard).unwrap();
+                        exit_handler.on_exit(async move {
+                            tokio::task::spawn_blocking(move || drop(trace_writer_guard));
+                        });
```

Comment:
What happens on a "normal" exit? Does it also run this `on_exit` code in that case? Looks like it only runs it on `ctrl+c`?

Reply:
You're right, we don't have a way of handling this yet. This isn't a problem for the "normal" next.js flow where JS controls the exit.